app/domain/authentication/util/fetch_authenticator_secrets.rb

```diff
@@ -6,25 +6,35 @@ module Util
 
     FetchAuthenticatorSecrets = CommandClass.new(
       dependencies: {
-        fetch_secrets: ::Conjur::FetchRequiredSecrets.new
+        fetch_required_secrets: ::Conjur::FetchRequiredSecrets.new,
+        fetch_optional_secrets: ::Conjur::FetchOptionalSecrets.new,
+        optional_variable_names: []
       },
       inputs: %i[conjur_account authenticator_name service_id required_variable_names]
     ) do
       def call
-        @required_variable_names.each_with_object({}) do |variable_name, secrets|
-          full_variable_name     = full_variable_name(variable_name)
-          secrets[variable_name] = required_secrets[full_variable_name]
-        end
+        secret_map_for(@required_variable_names, required_secrets).merge(secret_map_for(@optional_variable_names, optional_secrets))
       end
 
       private
 
       def required_secrets
-        @required_secrets ||= @fetch_secrets.(resource_ids: required_resource_ids)
+        @required_secrets ||= @fetch_required_secrets.(resource_ids: resource_ids_for(@required_variable_names))
+      end
+
+      def optional_secrets
+        @optional_secrets ||= @fetch_optional_secrets.(resource_ids: resource_ids_for(@optional_variable_names))
+      end
+
+      def secret_map_for(variable_names, secret_values)
+        variable_names.each_with_object({}) do |variable_name, secrets|
+          full_variable_name     = full_variable_name(variable_name)
+          secrets[variable_name] = secret_values[full_variable_name]
+        end
```

Comment:
I think the `secret_values` hash should have all the info we need here:
```suggestion
      def secret_map_for(secret_values)
        secret_values.each_with_object({}) do |(full_name, value), secrets|
          short_name = full_name.to_s.split('/')[-1]
          secrets[short_name] = value
        end
```